drivers/storage/portworx/component/telemetry.go

```diff
@@ -0,0 +1,170 @@
+	// TelemetryConfigMapName is the name of the config map that stores the telemetry configuration for CCM
+	TelemetryConfigMapName = "px-telemetry-config"
+	// TelemetryPropertiesFilename is the name of the CCM properties file
+	TelemetryPropertiesFilename = "ccm.properties"
+	// TelemetryArcusLocationFilename is name of the file storing the location of arcus endpoint to use
+	TelemetryArcusLocationFilename = "location"
+)
+
+type telemetry struct {
+	isCreated bool
+	k8sClient client.Client
+}
+
+func (t *telemetry) Name() string {
+	return TelemetryComponentName
+}
+
+func (t *telemetry) Priority() int32 {
+	return DefaultComponentPriority
+}
+
+func (t *telemetry) Initialize(k8sClient client.Client, _ version.Version, _ *runtime.Scheme, _ record.EventRecorder) {
+	t.k8sClient = k8sClient
+}
+
+func (t *telemetry) IsEnabled(cluster *corev1.StorageCluster) bool {
+	return cluster.Spec.Telemetry != nil && cluster.Spec.Telemetry.Enabled
+}
+
+func (t *telemetry) Reconcile(cluster *corev1.StorageCluster) error {
+	ownerRef := metav1.NewControllerRef(cluster, pxutil.StorageClusterKind())
+	if err := t.createConfigMap(cluster, ownerRef); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (t *telemetry) Delete(cluster *corev1.StorageCluster) error {
+	ownerRef := metav1.NewControllerRef(cluster, pxutil.StorageClusterKind())
+	if err := k8sutil.DeleteConfigMap(t.k8sClient, TelemetryConfigMapName, cluster.Namespace, *ownerRef); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (t *telemetry) MarkDeleted() {
+	t.isCreated = false
+}
+
+// RegisterTelemetryComponent registers the telemetry  component
+func RegisterTelemetryComponent() {
+	Register(TelemetryComponentName, &telemetry{})
+}
+
+func (t *telemetry) createConfigMap(
+	cluster *corev1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+) error {
+	config := fmt.Sprintf(`{
+      "product_name": "portworx",
+       "logging": {
+         "array_info_path": "/var/log/array_info.json"
+       },
+       "features": {
+         "appliance_info": "config",
+         "cert_store": "k8s",
+         "config_reload": "file",
+         "env_info": "file",
+         "scheduled_log_uploader":"enabled",
+         "upload": "enabled"
+       },
+      "cert": {
+        "activation": {
+              "private": "/dev/null",
+              "public": "/dev/null"
+        },
+        "registration_enabled": "true",
+        "appliance": {
+          "current_cert_dir": "/etc/pwx/ccm/cert"
+        }
+      },
+      "k8s": {
+        "cert_secret_name": "pure-telemetry-certs",
+        "cert_secret_namespace": "%s"
+     },
+     "cloud": {
+       "array_loc_file_path": "/etc/ccm/%s"
+     },
+      "server": {
+        "hostname": "0.0.0.0"
+      },
+      "logupload": {
+        "logfile_patterns": [
+            "/var/cores/*diags*",
+            "/var/cores/*px-cores*",
+            "/var/cores/*.heap",
+            "/var/cores/*.stack",
+            "/var/cores/.alerts/alerts*"
```

Comment:
Add `/var/cores/*.log` here instead of as additional_files ?

Reply:
anything in `logfile_patterns` expects these files to end with a specific format. The files need to be gzipped and have the ‘YYYYMMDD’ in the filename.

Hence the *.log files haven't been kept in additional_files.